cdap-api/src/main/java/co/cask/cdap/api/artifact/PluginContext.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.artifact;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.templates.AdapterConfigurer;
+import co.cask.cdap.api.templates.ApplicationTemplate;
+import co.cask.cdap.api.templates.plugins.PluginConfig;
+import co.cask.cdap.api.templates.plugins.PluginProperties;
+
+/**
+ * Provides access to plugin context when a program is executing.
+ */
+@Beta
+public interface PluginContext {
+
+  /**
+   * Gets the {@link PluginProperties} associated with the given plugin type and name.
+   *
+   * @param pluginId the unique identifier provide when declaring plugin usage in the program.
+   * @return the {@link PluginProperties}.
+   * @throws IllegalArgumentException if no plugin for the given type and name
+   * @throws UnsupportedOperationException if the program does not support plugin
+   */
+  PluginProperties getPluginProps(String pluginId);
```

Comment:
Since AbstractContext now needs to implement both AdapterContext and PluginContext, I had to change the name of the method in PluginContext. Will add a JIRA/TODO to change this method once we remove templates/adapters.